static/js/services/enketo.js

```diff
@@ -207,9 +207,17 @@ angular.module('inboxServices').service('Enketo',
           if (!summary) {
             return;
           }
+
+          var xmlStr;
+          try {
+            xmlStr = json2xml({ context: summary.context });
+          } catch (e) {
+            xmlStr = json2xml({ context: [] });
+          }
```

Comment:
I worry that this will catch too much. Is the error complaining about `context` being null? If so, can we just pass...

```js
json2xml({ context: summary.context || [] });
```

This works around this specific issue without calling the potentially expensive `json2xml` function twice, and without catching more errors than you intended.

Reply:
Unfortunately, the code breaks on line 44 of: https://github.com/estheban/node-json2xml/blob/master/lib/json2xml.js . It recurses over all the json object .. and !!undefined will always trigger an error.

Reply:
Ok. Looking at the [configuration documentation for `context`](https://github.com/medic/medic-docs/blob/master/configuration/contact-summary.md#context), it should be an object not an array. Given that it should not go down the array branch and should be handled by line 54 where it sets that property to `''`.

I don't think we should try and handle misconfigurations like returning an array instead of an object because there are infinitely many ways that could be broken. Instead I suggest you catch the error like you are, and then just $log.error it so whoever is testing the configuration gets a more useful message, and then exit the function.

If you can write a test which explodes even when given an object for the context then that's something we should consider handling.

Reply:
I have switched from returning an empty array (sorry about that) to throwing an error, I also added the additional test you requested - with context being an object which contains an [undefined] element.